File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        
        int maxLength=0;
        
        
        for(int right=0,left=0;right<s.length();right++){
            
            int firstApperanceIndex=s.indexOf(s.charAt(right),left);
            
            if(firstApperanceIndex!=right){
                
                left=firstApperanceIndex+1;
                
            }
             maxLength=Math.max(maxLength,right-left+1);
        }
        
       
        
        return maxLength;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        maxLen=0
        left=0
        for right in range(len(s)):
            firstIndex=s.index(s[right],left)
            if firstIndex!=right:
                left=firstIndex+1
            maxLen=max(maxLen,right-left+1)



        
        return maxLen;